master/pkg/model/task.go

```diff
@@ -1,32 +1,48 @@
 package model
 
-import "time"
+import (
+	"time"
 
-// TaskID is the unique ID of a task among all tasks.
-type TaskID string
+	"github.com/google/uuid"
 
-// TaskType is the type of a task.
-type TaskType string
+	"github.com/determined-ai/determined/proto/pkg/taskv1"
+)
 
-// AllocationID is the ID of an allocation of a task. It is usually of the form
-// TaskID.allocation_number, maybe with some other metadata if different types of
-// allocations run.
-type AllocationID string
+// JobID is the unique ID of a job among all jobs.
+type JobID string
 
-// NewAllocationID returns a new unique task id.
-func NewAllocationID(name string) AllocationID {
-	return AllocationID(name)
+// NewJobID returns a random, globally unique job ID.
+func NewJobID() JobID {
+	return JobID(uuid.New().String())
 }
 
-// Allocation is the model for an allocation in the database.
-type Allocation struct {
-	TaskID       TaskID       `db:"task_id"`
-	AllocationID AllocationID `db:"allocation_id"`
-	ResourcePool string       `db:"resource_pool"`
-	StartTime    time.Time    `db:"start_time"`
-	EndTime      *time.Time   `db:"end_time"`
+// JobType is the type of a job.
+type JobType string
```

Comment:
what is the plan for this?

Reply:
I think at some point they have to be saved to the DB with all the information that is share among all jobs.
E.G. all jobs, since they are a user submission by definition, have a owner_id, but now that is not persisted, except for experiments, so we can't include it in historical resource allocation. I think I will create public.jobs in a follow up and save stuff like owner_id in it, maybe some others, to include it in aggs properly.